src/js/components/modals/ServiceFormModal.js

```diff
@@ -0,0 +1,240 @@
+import Ace from 'react-ace';
+import mixin from 'reactjs-mixin';
+import {Modal} from 'reactjs-components';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import 'brace/mode/json';
+import 'brace/theme/monokai';
+import 'brace/ext/language_tools';
+
+import MarathonStore from '../../stores/MarathonStore';
+import SchemaForm from '../SchemaForm';
+import Service from '../../structs/Service';
+import ServiceUtil from '../../utils/ServiceUtil';
+import ServiceSchema from '../../constants/ServiceSchema';
+import ToggleButton from '../ToggleButton';
+
+const METHODS_TO_BIND = [
+  'getAdvancedSubmit',
+  'handleCancel',
+  'handleClearError',
+  'handleJSONChange',
+  'handleJSONToggle',
+  'handleSubmit',
+  'onMarathonStoreServiceCreateError',
+  'onMarathonStoreServiceCreateSuccess'
+];
+
+class ServiceFormModal extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    let model =
+      ServiceUtil.createFormModelFromSchema(ServiceSchema);
+    this.state = {
+      app: JSON.stringify({id:'', cmd:''}, null, 2),
```

Comment:
Now that we call everything `service`, `app` is just confusing. Let's name this one `jsonDefinition` or similar, as its sole purpose is to hold the JSON mode blob.